aiida_nanotech_empa/workflows/gaussian/delta_scf_workchain.py

```diff
@@ -0,0 +1,267 @@
+from aiida_nanotech_empa.workflows.gaussian import common
+
+from aiida.engine import WorkChain, calcfunction, ExitCode
+from aiida.orm import Int, Str, Code, Dict
+from aiida.orm import StructureData, RemoteData
+
+from aiida.plugins import WorkflowFactory
+
+GaussianBaseWorkChain = WorkflowFactory('gaussian.base')
+
+
+@calcfunction
+def subtract(a, b):
+    return a - b
+
+
+class GaussianDeltaScfWorkChain(WorkChain):
+    @classmethod
+    def define(cls, spec):
+        super().define(spec)
+
+        spec.input("gaussian_code", valid_type=Code)
+
+        spec.input('structure',
+                   valid_type=StructureData,
+                   required=True,
+                   help='input geometry')
+        spec.input('functional',
+                   valid_type=Str,
+                   required=True,
+                   help='xc functional')
+
+        spec.input('basis_set',
+                   valid_type=Str,
+                   required=True,
+                   help='basis_set')
+
+        spec.input('multiplicity',
+                   valid_type=Int,
+                   required=False,
+                   default=lambda: Int(1),
+                   help='spin multiplicity; 0 means RKS')
+
+        spec.input(
+            "parent_calc_folder",
+            valid_type=RemoteData,
+            required=False,
+            help="the folder of a completed gaussian calculation",
+        )
+
+        spec.outline(cls.setup, cls.submit_scfs, cls.finalize)
+
+        spec.outputs.dynamic = True
+
+        spec.exit_code(
+            301,
+            "ERROR_MULTIPLICITY",
+            message="Multiplicity and number of el. doesn't match.",
+        )
+        spec.exit_code(
+            390,
+            "ERROR_TERMINATION",
+            message="One or more steps of the work chain failed.",
+        )
+
+    def is_uks(self):
+        if self.inputs.multiplicity.value == 0:
+            return False
+        return True
+
+    def setup(self):
+        self.report("Inspecting input and setting up things")
+
+        pymatgen_structure = self.inputs.structure.get_pymatgen_molecule()
+        self.ctx.n_atoms = pymatgen_structure.num_sites
+        n_electrons = pymatgen_structure.nelectrons
+
+        if self.is_uks():
+            self.ctx.functional = 'u' + self.inputs.functional.value
+            self.ctx.mult = self.inputs.multiplicity.value
+        else:
+            self.ctx.functional = self.inputs.functional.value
+            self.ctx.mult = 1
+
+        if self.ctx.mult % 2 == n_electrons % 2:
```

Comment:
The part that determines multiplicity - can it be moved to the `utils` folder? I think it is a pretty useful tool.

Reply:
well at the moment it just checks if the input multiplicity and the number of electrons are compatible. Not very general, as I don't check the charge (but in the workflows I don't really allow for ion calculations anyways). I do use this same piece of code in many of the workflows so I will separate this into `gaussian.workflows.common`

Reply:
You can add a charge as an additional parameter that has a 0 default. Again, it is not very specific to the gaussian.